devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -0,0 +1,132 @@
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap && descMap.clusterInt == 0x0006) {
+			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+				if (device.currentValue("level") == 0) {
+					sendEvent(name: "level", value: STEP)
+				}
+				sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
+			}
+		} else if (descMap && descMap.clusterInt == 0x0008) {
+			def currentLevel = device.currentValue("level") as Integer ?: 0
+			if (descMap.commandInt == 0x02) {
+				def value = Math.min(currentLevel + STEP, 100)
+				log.debug "move to ${descMap.data}"
+				if (descMap.data[0] == "00") {
+					log.debug "move up"
+					sendEvent(name: "switch", value: "on")
+					sendEvent(name: "level", value: value)
+				} else if (descMap.data[0] == "01") {
+					log.debug "move down"
+					value = Math.max(currentLevel - STEP, 0)
+					// don't change level if switch will be turning off
+					if (value == 0) {
+						sendEvent(name: "switch", value: "off")
+					} else {
+						sendEvent(name: "level", value: value)
+					}
+				}
+			} else if (descMap.commandInt == 0x01) {
+				sendEvent(name: "level", value: descMap.data[0] == "00" ? 100 : STEP)
+				sendEvent(name: "switch", value: "on" )
+				log.debug "step to ${descMap.data}"
+			} else if (descMap.commandInt == 0x03) {
+				log.debug "stop move"
+			}
+		} else if (descMap && descMap.clusterInt == 0x0005) {
+			if (descMap.commandInt == 0x05) {
+				sendEvent(name: "button", value: "pressed", isStateChange: true)
+			} else if (descMap.commandInt == 0x04) {
+				sendEvent(name: "button", value: "held", isStateChange: true)
+			}
+		} else {
+			log.warn "DID NOT PARSE MESSAGE for description : $description"
+			log.debug "${descMap}"
+		}
+	}
+}
+
+def off() {
+	sendEvent(name: "switch", value: "off", isStateChange: true)
+}
+
+def on() {
+	sendEvent(name: "switch", value: "on", isStateChange: true)
+}
+
+def setLevel(value) {
+	if (value == 0) {
+		sendEvent(name: "switch", value: "off")
+	} else {
+		sendEvent(name: "switch", value: "on")
+        sendEvent(name: "level", value: value)
+	}
+}
+
+def installed() {
+	sendEvent(name: "switch", value: "on", isStateChange: false, displayed: false)
+	sendEvent(name: "level", value: 100, isStateChange: false, displayed: false)
+	sendEvent(name: "button", value: "pressed", isStateChange: false, displayed: false)
+}
+
+def configure() {
+    // strangely, these are necessary to have the device report when its buttons are pressed
+	zigbee.addBinding(0x0006) + zigbee.addBinding(0x0008) + zigbee.addBinding(0x0005)
+//	zigbee.onOffConfig() + zigbee.levelConfig() + zigbee.configureReporting(0x0005, 0x0001, DataType.UINT8, 1, 3600, null)
```

Comment:
You can remove this commented line.  You could also change the above to use `zigbee.ONOFF_CLUSTER`, `zigbee.LEVEL_CONTROL_CLUSTER` for the first 2.  We don't have a constant currently for the scenes cluster though.